types/next/index.d.ts

```diff
@@ -71,7 +72,11 @@ declare module 'next/head' {
   import * as React from 'react';
 
   function defaultHead(): JSX.Element[];
-  export default class extends React.Component<{}, {}> {}
+  export default class extends React.Component<{}, {}> {
+    static canUseDOM: boolean;
+    static peek(): Array<React.ReactElement<any>>;
+    static rewind(): Array<React.ReactElement<any>>;
+  }
 }
 
 declare module 'next/document' {
```

Comment:
Instead of using `declare module`, these could each be there own file, e.g. `next/document.d.ts`.